slices/reduce.go

```diff
@@ -0,0 +1,25 @@
+package slices
+
+import "golang.org/x/exp/constraints"
+
+func ReduceWith[S ~[]E, E any, A any](s S, acc A, fn func(A, E) A) A {
+	for _, e := range s {
+		acc = fn(acc, e)
+	}
+
+	return acc
+}
+
+func Reduce[S ~[]E, E any, A any](s S, fn func(A, E) A) A {
+	var acc A
+
+	return ReduceWith(s, acc, fn)
+}
+
+type sumable interface {
+	constraints.Integer | constraints.Float | ~string
+}
+
+func Sum[T sumable](i, j T) T {
```

Comment:
```suggestion
func Sum[T sumable](vs ...T) T {
```

you could generalize it

Reply:
```suggestion
func Sum[T sumable](values ...T) T {
```

Reply:
That won't compile unless the reduce function accepts variadic args too, which kind of defeats the purpose of a reduce.

We could change things up and have `Sum` call Reduce which will provides a function link `sum := slices.Sum(numbers)`